lib/spack/spack/database.py

```diff
@@ -579,23 +626,33 @@ def _read(self):
                 self._write(None, None, None)
             self.reindex(spack.store.layout)
 
-    def _add(self, spec, directory_layout=None, explicit=False):
+    def _add(self, spec, directory_layout=None, **kwargs):
```

Comment:
why not keep the arguments explicitly?

Reply:
see below answer

Reply:
Here it is a bit different, how do I know why kwargs are accepted/expected? And documentation ~~can~~ will become outdated... 
I agree with you that writing out arguments makes a call easier to read, but I also know that I always have to google the damn kwargs of any matplotlib call... The former could be enforced by convention (which will probably not hold in the long run either). So I see points for both sides -> no strong feelings.

Reply:
In this case `_add` is by convention a *private* (if you pass the term in python) method. Usually I regard those methods as implementation details and don't document them using the same care as I have for user facing APIs. 

If you think it should be documented with the same level of details as `query` that won't be a problem - just let me know. In any case users are not supposed to be calling `_add`, and the function itself won't appear in the docs online.

Reply:
Extended help here.

Reply:
can this one be `explicit=False` again too?